### PR TITLE
Add LICENSE.md for error-stack

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

Releases `error-stack` under an MIT license.

## 🔍 What does this change?

- Add *LICENSE.md*
- Mention the license on *Cargo.toml* (required for releasing)

### CODE REVIEW

**Review by Alfred-Mountfield — CHANGES_REQUESTED** [review]

**Review by Alfred-Mountfield — CHANGES_REQUESTED** [review]
As expressed off Slack I think we should re-work the layout of this README heavily to guide people more towards quick-start usage, and moving specifics into supplementary files. 

I also think we should spend less time comparing with other libraries and justifying design decisions in the README, making sure that we make that capture that information elsewhere (docs, supplementary files and/or by pointing people to the announcement)

File: packages/libs/error-stack/README.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+We haven’t made it easy to create a new `Report` from a `String` because it requires the type to implement `Context`. It’s still possible to use a newtype `UniqueError(String)` if a user wants to, but we discourage this usage because we’ve found that a lot of those places benefit from attaching a string message on a well-defined context instead. In any case, other libraries most probably still have to change the context from `UniqueError` to their context because of the explicit type requirement.
+
+**By consciously imposing more restrictions in some areas we hope to reduce the overall cost of development, in keeping with the philosophy underpinning Rust itself.**
+
+## What else `error-stack` is offering?
+
+Having the concept of `error-stack` explain, these are the functionalities we also provide. Don't forget to checkout the [documentation] as well!
```
Comment (nonparibus) [inline packages/libs/error-stack/README.md:130]: The first sentence here reads incorrectly; unsure of best/exact intended phrasing.

File: packages/libs/error-stack/README.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+   3: parse_config
+             at main.rs:4:19
+   4: main
+             at main.rs:13:26
+   5: ...
+
+Suggestion: Use a file you can read next time!
+```
+
+## What’s the catch?
```
[the remark this answers is not in the document]
Comment (nonparibus) [reply packages/libs/error-stack/README.md:118]: I don't object to keeping some sort of 'Considerations' section in the main README, which I think done a bit more broadly -- as opposed to focused solely on performance -- would be helpful reading to anyone considering adopting the library.

**Comment by nonparibus** [comment]
I agree with the part of Alfie's top-level review comment which states we should link to the announcement post for discussion of alternative solutions/additional contextual information.

However I would not split 'Performance' out into another file -- instead making it more concise and folding it into a 'Considerations' section containing things potential adopters should consider.

**Comment by TimDiekmann** [comment]
Removed README.md in favor of #587

**Review by Alfred-Mountfield — APPROVED** [review]

**Review by nonparibus — APPROVED** [review]

**Pull request merged**